### DISCUSSION TITLE
Allow customizing font weight

### DISCUSSION THREAD

**Discussion by frolic** [rainbow-me/rainbowkit#361] [Ideas]
I adjusted my theme to use Space Mono but everything being bold by default makes it a little hard to read. Is it possible to allow customizing the font weight?

Edit: Looks like this is mostly because Space Mono only comes in two weights, but RainbowKit seems to use several different ones, so most things were appearing ~bold.

**Comment by er1chu** [comment]
Seconding here, the low level theme API is already really useful but really need this one last thing at the moment.

**Comment by jjjjjjjjjjjjjjjjacob** [comment]
thirding. it's amazing how quick this library is to set up-- major props to the developers & the team at Rainbow (& wagmi). But it would be helpful to have more font customization.

**Comment by peduarte** [comment]
hi there, starting to look into this. but curious about how much control do you want?

we're exploring a couple of different options:

**1. choose from a predefined font-weight option**
```jsx
lightTheme({
  fontWeight: 'bold'
})
```

where `fontWeight` could potentially be: `light|regular|bold`

this is inline with `fontStack` and `borderRadius` options and it gives users an easy and quick way to change the overall feel of the font weight. rainbowkit internally would choose which font weight to set where, based on font sizes. 

**2. expose a font-weight theme token for every text** 
```js
const myCustomTheme: Theme = {
  fontWeights: {
    connectButtonText: 'medium',
    connectButtonTextError: 'bold',
    modalText: '...',
    modalTextDim: '...',
    modalTextSecondary: '...',
    standby: '',
    ...probablyMore
  }
}
```

This gives you more granular control, but also it's more of a hassle to set up.

**3. both of the above**

we could also do both, as we do for `accentColor`, `accentColorForeground` and `borderRadius`

**4. expose font-weight theme tokens but limited to different weights**

```js
const myCustomTheme: Theme = {
  fontWeights: {
    light: '100',
    regular: '400',
    bold: '700'
  }
}
```

with this approach, rainbowkit internally would still choose where to apply which weight, but users have the option to define them

**Reply by jacobedawson** [reply]
I think option 1 would be fine to begin with, over time more options would be great but not critical

**Reply by peduarte** [reply]
cool cool. out of curiosity, did you override the font stack in your app? if so, which font are you using?

**Comment by dylanngph** [comment]
Hi, the font-weight for new version is gone, how can i access now?

**Comment by mihoward21** [comment]
Did this ever get implemented? I'm using the modal with my own connect button, but would love to be able to customize the font weights in that modal. Any of those options you mentioned @peduarte would be great. Happy to use predefined options or have it be more customizable, whatever is easier for you all to get shipped.

I'm using: `@rainbow-me/rainbowkit 1.3.1`
